src/wallet/db.cpp

```diff
@@ -137,4 +138,12 @@ bool IsSQLiteFile(const fs::path& path)
     // Check the application id matches our network magic
     return memcmp(Params().MessageStart(), app_id, 4) == 0;
 }
+
+void ReadDatabaseArgs(const ArgsManager& args, DatabaseOptions& options)
+{
+    if (args.IsArgSet("-unsafesqlitesync")) options.use_unsafe_sync = args.GetBoolArg("-unsafesqlitesync", options.use_unsafe_sync);
```

Comment:
Why is the IsArgSet check needed? Looks like this either makes the default argument path unused, or the default argument path makes this redundant.

Reply:
> Why is the IsArgSet check needed? Looks like this either makes the default argument path unused, or the default argument path makes this redundant.

That's true. I think I suggested IsArgSet to try to make it clear that all the current option values would be unchanged if they weren't explicitly overridden by ArgsManager. But probably better to simplify. Maybe:

```c++
// Override current options with args values, if any were specified
options.use_unsafe_sync = args.GetBoolArg("-unsafesqlitesync", options.use_unsafe_sync);
options.use_shared_memory = !args.GetBoolArg("-privdb", !options.use_shared_memory);
options.max_log_mb = args.GetIntArg("-dblogsize", options.max_log_mb);
```

Reply:
I have modified it as ryanofsky suggested.